### ISSUE TITLE
java.lang.NullPointerException when use TIMEDIFF

### ISSUE THREAD

**Issue by Dracks977** [metabase/metabase#10983]
**Describe the bug**
when i use native query or view i got java.lang.NullPointerException
**Logs**
``` JSON
{"database_id":2,"started_at":"2019-09-25T08:59:06.813Z","json_query":{"database":2,"type":"native","native":{"query":"select healthcare_requests.id, TIMEDIFF(sms_logs.updated_at, healthcare_requests.created_at) from healthcare_requests INNER JOIN sms_logs ON sms_logs.healthcare_request_id = healthcare_requests.id AND answer = true ORDER BY healthcare_requests.id DESC","template-tags":{}},"parameters":[],"async?":true,"middleware":{"add-default-userland-constraints?":true,"userland-query?":true},"constraints":{"max-results":10000,"max-results-bare-rows":2000}},"status":"failed","stacktrace":[null],"context":"ad-hoc","error":"java.lang.NullPointerException","row_count":0,"running_time":755,"data":{"rows":[],"cols":[]}}
```

``` SQL
select `healthcare_requests`.`id` AS `healthcare_requests_id`,timediff(`sms_logs`.`updated_at`,`healthcare_requests`.`created_at`) AS `DateDiff` from (`healthcare_requests` join `sms_logs` on(((`sms_logs`.`healthcare_request_id` = `healthcare_requests`.`id`) and (`sms_logs`.`answer` = TRUE)))) order by `healthcare_requests`.`id` desc
```

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

- Your operating system: debian 9
- Your databases: mariadb
- Metabase version:  0.32.10
- Metabase hosting environment: docker
- Metabase internal database: H2 

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by stephanvd** [comment]
Also seeing this bug when selecting a timediff. Worked around it for now by wrapping it in an empty concat:
`concat(timediff(uploader_start_timestamp, uploader_end_timestamp), '')`

**Comment by flamber** [comment]
Just to confirm on 0.33.6 with MariaDB 10.4 and MySQL 5.7:
`SELECT timediff(current_date + INTERVAL 1 HOUR, current_timestamp)`
```
11-20 13:33:54 WARN middleware.process-userland-query :: Query failure {:status :failed,
 :class java.lang.NullPointerException,
 :error "java.lang.NullPointerException",
 :stacktrace
 ("--> driver.mysql$eval77726$fn__77727.invoke(mysql.clj:333)"
  "driver.sql_jdbc.execute$eval76399$fn__76400$fn__76401$iter__76402__76406$fn__76407$fn__76408.invoke(execute.clj:121)"
  "driver.sql_jdbc.execute$eval76399$fn__76400$fn__76401$iter__76402__76406$fn__76407.invoke(execute.clj:120)"
  "driver.sql_jdbc.execute$cancelable_run_query$fn__76438.invoke(execute.clj:182)"
  "driver.sql_jdbc.execute$do_with_ensured_connection.invokeStatic(execute.clj:159)"
  "driver.sql_jdbc.execute$do_with_ensured_connection.invoke(execute.clj:157)"
  "driver.sql_jdbc.execute$cancelable_run_query.invokeStatic(execute.clj:174)"
  "driver.sql_jdbc.execute$cancelable_run_query.invoke(execute.clj:170)"
  "driver.sql_jdbc.execute$run_query.invokeStatic(execute.clj:196)"
  "driver.sql_jdbc.execute$run_query.invoke(execute.clj:192)"
  "driver.sql_jdbc.execute$do_with_auto_commit_disabled.invokeStatic(execute.clj:243)"
  "driver.sql_jdbc.execute$do_with_auto_commit_disabled.invoke(execute.clj:231)"
  "driver.sql_jdbc.execute$do_in_transaction$fn__76459.invoke(execute.clj:248)"
  "driver.sql_jdbc.execute$do_in_transaction.invokeStatic(execute.clj:247)"
  "driver.sql_jdbc.execute$do_in_transaction.invoke(execute.clj:246)"
  "driver.sql_jdbc.execute$run_query_without_timezone.invokeStatic(execute.clj:265)"
  "driver.sql_jdbc.execute$run_query_without_timezone.invoke(execute.clj:264)"
  "driver.sql_jdbc.execute$execute_query$fn__76486.invoke(execute.clj:297)"
  "driver.sql_jdbc.execute$do_with_try_catch.invokeStatic(execute.clj:223)"
  "driver.sql_jdbc.execute$do_with_try_catch.invoke(execute.clj:217)"
  "driver.sql_jdbc.execute$execute_query.invokeStatic(execute.clj:291)"
  "driver.sql_jdbc.execute$execute_query.invoke(execute.clj:285)"
  "driver.sql_jdbc$eval76784$fn__76785.invoke(sql_jdbc.clj:44)"
  "query_processor$eval51161$execute_query__51166$fn__51167.invoke(query_processor.clj:71)"
  "query_processor$eval51161$execute_query__51166.invoke(query_processor.clj:65)"
  "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__40382.invoke(mbql_to_native.clj:38)"
  "query_processor.middleware.annotate$result_rows_maps__GT_vectors$fn__42744.invoke(annotate.clj:541)"
  "query_processor.middleware.annotate$add_column_info$fn__42650.invoke(annotate.clj:485)"
  "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__43793.invoke(cumulative_aggregations.clj:57)"
  "query_processor.middleware.resolve_joins$resolve_joins$fn__47861.invoke(resolve_joins.clj:184)"
  "query_processor.middleware.limit$limit$fn__44448.invoke(limit.clj:19)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__50974.invoke(results_metadata.clj:87)"
  "query_processor.middleware.format_rows$format_rows$fn__44428.invoke(format_rows.clj:26)"
  "query_processor.middleware.add_dimension_projections$add_remapping$fn__41167.invoke(add_dimension_projections.clj:232)"
  "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__41846.invoke(add_source_metadata.clj:107)"
  "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__47919.invoke(resolve_source_table.clj:46)"
  "query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__41679.invoke(add_row_count_and_status.clj:16)"
  "query_processor.middleware.driver_specific$process_query_in_context$fn__43891.invoke(driver_specific.clj:12)"
  "query_processor.middleware.add_settings$add_settings$fn__41710.invoke(add_settings.clj:45)"
  "query_processor.middleware.resolve_driver$resolve_driver$fn__47509.invoke(resolve_driver.clj:22)"
  "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__43097$fn__43098.invoke(bind_effective_timezone.clj:9)"
  "util.date$call_with_effective_timezone.invokeStatic(date.clj:88)"
  "util.date$call_with_effective_timezone.invoke(date.clj:77)"
  "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__43097.invoke(bind_effective_timezone.clj:8)"
  "query_processor.middleware.store$initialize_store$fn__51013$fn__51014.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:46)"
  "query_processor.store$do_with_store.invoke(store.clj:40)"
  "query_processor.middleware.store$initialize_store$fn__51013.invoke(store.clj:10)"
  "query_processor.middleware.async$async__GT_sync$fn__40277.invoke(async.clj:23)"
  "query_processor.middleware.async_wait$runnable$fn__42809.invoke(async_wait.clj:89)"),
 :query
 {:type "native",
  :native {:query "SELECT timediff(current_date + INTERVAL 1 HOUR, current_timestamp)", :template-tags {}},
  :parameters [],
  :async? true,
  :middleware {:add-default-userland-constraints? true, :userland-query? true},
  :info
  {:executed-by 1,
   :context :ad-hoc,
   :card-id nil,
   :nested? false,
   :query-hash [62, -96, -21, 18, -92, 85, 28, -82, 91, -108, 7, 68, -64, 37, -97, -104, 89, 87, -11, 57, 98, -16, 44, 78, -102, 41, -115, 33, 66, 36, 20, 70]},
  :constraints {:max-results 10000, :max-results-bare-rows 2000}}}

11-20 13:33:54 DEBUG middleware.log :: POST /api/dataset 200 [ASYNC: completed] 164.9 ms (10 DB calls) (174 total active threads) Queries in flight: 0
```

**Comment by theoaf** [comment]
Also seeing this same issue with TIMEDIFF on mariadb version 10.3.8 with the latest version of metabase

**Comment by camsaul** [comment]
Fixed by #12447
